WireUI/Tests/WireMainNavigationUITests/Containers/MainTabBarControllerTests.swift

```diff
@@ -23,67 +23,72 @@ import XCTest
+        MockConversationListViewController,
+        MockConversationViewController<PreviewConversationModel>
+    >!
     private var snapshotHelper: SnapshotHelper!
 
-    override func setUp() {
+    @MainActor
+    override func setUp() async throws {
+        sut = .init()
         snapshotHelper = .init()
             .withSnapshotDirectory(SnapshotTestReferenceImageDirectory)
     }
 
     override func tearDown() {
         snapshotHelper = nil
+        sut = nil
     }
 
     @MainActor
-    func testConversationListAndConversationIsInstalled() throws {
+    func testConversationListIsInstalled() throws {
         // Given
         let conversationList = MockConversationListViewController()
-        let conversation = UIViewController()
-        let sut = MainTabBarController<MockConversationListViewController>()
 
         // When
-        sut.conversations = (conversationList, conversation)
+        sut.conversationList = conversationList
 
         // Then
-        let navigationController = try XCTUnwrap(sut.viewControllers?[1] as? UINavigationController)
-        XCTAssertEqual(navigationController.viewControllers, [conversationList, conversation])
+        let navigationController = try XCTUnwrap(sut.viewControllers?[1] as? UINavigationController) // TODO: [WPB-6647] fix index for navigation overhaul
+        XCTAssertEqual(navigationController.viewControllers, [conversationList])
     }
 
     @MainActor
-    func testConversationIsReleased() throws {
+    func testConversationIsReleased() async throws {
         // Given
-        let sut = MainTabBarController<MockConversationListViewController>()
-        sut.conversations = (.init(), .init())
-        let navigationController = try XCTUnwrap(sut.viewControllers?[1] as? UINavigationController)
+        weak var weakConversationList: MockConversationListViewController?
+        sut.conversationList = {
+            let conversationList = MockConversationListViewController()
+            weakConversationList = conversationList
+            return conversationList
+        }()
 
         // When
-        navigationController.popViewController(animated: false)
+        sut.conversationList = nil
 
         // Then
-        let (conversationList, conversation) = try XCTUnwrap(sut.conversations)
-        XCTAssertNotNil(conversationList)
-        XCTAssertNil(conversation)
+        await Task.yield()
+        XCTAssertNil(weakConversationList)
     }
 
     @MainActor
     func testArchiveIsInstalled() throws {
         // Given
         let archive = UIViewController()
-        let sut = MainTabBarController<MockConversationListViewController>()
 
         // When
         sut.archive = archive
 
         // Then
-        let navigationController = try XCTUnwrap(sut.viewControllers?[3] as? UINavigationController)
+        let navigationController = try XCTUnwrap(sut.viewControllers?[3] as? UINavigationController) // TODO: [WPB-6647] fix index for navigation overhaul
```

Comment:
is this todo tackled in a later PR?

Reply:
Yes, in the main PR for the epic branch: https://github.com/wireapp/wire-ios/pull/1361
On develop we still show four tabs.